src/components/common/ColonyActions/CountDownTimer/helpers.ts

```diff
@@ -0,0 +1,197 @@
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import {
+  GetMotionTimeoutPeriodsReturn,
+  useGetMotionTimeoutPeriodsQuery,
+} from '~gql';
+import { RefetchMotionState, useAppContext, useColonyContext } from '~hooks';
+import { MotionState } from '~utils/colonyMotions';
+
+import { useStakingWidgetContext } from '../ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget';
+
+const splitTimeLeft = (period: number) => {
+  if (period > 0) {
+    return {
+      days: Math.floor(period / (60 * 60 * 24)),
+      hours: Math.floor((period / (60 * 60)) % 24),
+      minutes: Math.floor((period / 60) % 60),
+      seconds: Math.floor(period % 60),
+    };
+  }
+  return undefined;
+};
+
+const getCurrentStatePeriodInMs = (
+  motionState: MotionState,
+  motionTimeoutPeriods: GetMotionTimeoutPeriodsReturn,
+) => {
+  switch (motionState) {
+    case MotionState.Staking:
+    case MotionState.Staked:
+    case MotionState.Objection:
+      return motionTimeoutPeriods.timeLeftToStake;
+    case MotionState.Voting:
+      return motionTimeoutPeriods.timeLeftToVote;
+    case MotionState.Reveal:
+      return motionTimeoutPeriods.timeLeftToReveal;
+    case MotionState.Escalation:
+      return motionTimeoutPeriods.timeLeftToEscalate;
+    default:
+      return '-1';
+  }
+};
+
+const useMotionTimeoutPeriods = (colonyAddress = '', motionId: number) => {
+  const { data, loading, refetch } = useGetMotionTimeoutPeriodsQuery({
+    variables: { input: { colonyAddress, motionId } },
+    notifyOnNetworkStatusChange: true,
+    fetchPolicy: 'network-only',
+  });
+  const motionTimeoutPeriods = data?.getMotionTimeoutPeriods || {
+    timeLeftToEscalate: '-1',
+    timeLeftToVote: '-1',
+    timeLeftToReveal: '-1',
+    timeLeftToStake: '-1',
+  };
+
+  return {
+    motionTimeoutPeriods,
+    loadingMotionTimeoutPeriods: loading,
+    refetchMotionTimeoutPeriods: refetch,
+  };
+};
+
+export const useMotionCountdown = (
+  state: MotionState,
+  motionId: number,
```

Comment:
This should be typed as a string to avoid overflow in the event it's a very large number.